### ISSUE TITLE
Error executing postinstall script

### ISSUE THREAD

**Issue by drewval** [jimkang/homophonizer#3]
Failed at the homophonizer@1.1.0 postinstall script 'make builddbs'.

**Comment by jimkang** [comment]
Was there an error message or any other details?

**Comment by drewval** [comment]
Thanks for the prompt response. That was pretty much the error.

I'm running `npm install homophonizer` in my project root folder which already has a node_modules folder - all my other packages are there.

I'm wondering if we need to cd into node_modules/homophonizer before running the `make builddbs` postscript? Is the cd already part of your postscript?

`npm ERR! homophonizer@1.1.0 postinstall:`make builddbs`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the homophonizer@1.1.0 postinstall script 'make builddbs'.
npm ERR! This is most likely a problem with the homophonizer package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     make builddbs
npm ERR! You can get their info via:
npm ERR!     npm owner ls homophonizer
npm ERR! There is likely additional logging output above.`

**Comment by jimkang** [comment]
Thanks for the details. Which Node and NPM versions are you using? It seems to get through it fine with Node 5.6 and NPM 3.6.

Also, do you have an `npm-debug.log`?

**Comment by drewval** [comment]
I have update my npm and node to the latest version but to no avail.

Has it got to do with the MSBuild version? I'm using 2015.

**Comment by jimkang** [comment]
Ah, I've never run it on Windows. On Unix and Mac, there's usually a `make` task runner on the system. Looks like [MinGW](http://www.mingw.org/wiki/Getting_Started) will install a compatible `make` on Windows. Can you see if installing that and making sure that it's in your PATH will help?

**Comment by drewval** [comment]
Right. 
I already have the latest version of make (GNU Make 3.81) for Windows installed (http://gnuwin32.sourceforge.net/packages/make.htm) and added to Environment Variable Path.
Is this different from MinGW?

**Comment by jimkang** [comment]
I have no idea which will work! But if you do `cd node_modules/homophonizer` then `make builddbs`, does anything happen?

**Comment by drewval** [comment]
If I cd into homophonizer and run `make builddbs` it will start building stuff.
However, I had to manually download your project from Github in order the run that instead of using `npm install`. It will then complain about dependencies.

Have you added a `cd node_modules/homophonizer` to the postinstall script before running `make builddbs`?

**Comment by jimkang** [comment]
The postinstall command normally runs in the working directory of the module that it's in. I tried it just now to make sure.

```
> npm install homophonizer                                                                            ⮂  
npm WARN deprecated queue-async@1.0.7: renamed to d3-queue
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN prefer global node-ninja@1.0.1 should be installed with -g
npm WARN prefer global node-gyp@3.3.1 should be installed with -g

> leveldown@1.4.6 install /Users/jimkang/gcw/check-on-your-upgrade/node_modules/leveldown
> prebuild --install


> homophonizer@1.1.0 postinstall /Users/jimkang/gcw/check-on-your-upgrade/node_modules/homophonizer
> make builddbs

cd metaphone && node buildmetaphonedb.js cmudict.0.7a-words-only.txt metaphone.db
cd phoneme && node buildphonemedb.js cmudict.0.7a phoneme.db
check-on-your-upgrade@1.0.0 /Users/jimkang/gcw/check-on-your-upgrade
└── homophonizer@1.1.0  extraneous
```

It completes the postinstall here.

But you seem to be seeing it run from the parent project's directory instead of `node_modules/homophonizer` on Windows?

I'll try to look into this some more tomorrow.

**Comment by drewval** [comment]
Yeah it seems to be running make from the project directory instead of the `node_modules/homophonizer` directory.
I'll keep at it but let me know if you have any breakthrough.
Thanks.

**Comment by drewval** [comment]
Hi @jimkang any updates to the module?
